org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -476,65 +477,22 @@ public void doGenerate(Resource resource, LFGeneratorContext context) {
         generateCodeFor(lfModuleName);
 
         // Derive target filename from the .lf filename.
-        var cFilename = CCompiler.getTargetFileName(lfModuleName, this.CCppMode);
+        var cFilename = CCompiler.getTargetFileName(lfModuleName, this.CCppMode, targetConfig);
         var targetFile = fileConfig.getSrcGenPath() + File.separator + cFilename;
 
         try {
 
+                String srcPrefix = targetConfig.platformOptions.platform == Platform.ARDUINO ? "src/" : "";
+
                 // Copy the core lib
                 FileUtil.copyDirectoryFromClassPath(
                     "/lib/c/reactor-c/core",
-                    fileConfig.getSrcGenPath().resolve("core"),
+                    fileConfig.getSrcGenPath().resolve(srcPrefix + "core"),
                     true
                 );
                 // Copy the C target files
                 copyTargetFiles();
 
-            // If we are running an Arduino Target, need to copy over the Arduino-CMake files.
-            if (targetConfig.platformOptions.platform == Platform.ARDUINO) {
-                FileUtil.copyDirectoryFromClassPath(
-                    "/lib/platform/arduino/Arduino-CMake-Toolchain/Arduino",
-                    fileConfig.getSrcGenPath().resolve("toolchain/Arduino"),
-                    false
-                );
-                FileUtil.copyDirectoryFromClassPath(
-                    "/lib/platform/arduino/Arduino-CMake-Toolchain/Platform",
-                    fileConfig.getSrcGenPath().resolve("toolchain/Platform"),
-                    false
-                );
-                FileUtil.copyFileFromClassPath(
-                    "/lib/platform/arduino/Arduino-CMake-Toolchain/Arduino-toolchain.cmake",
-                    fileConfig.getSrcGenPath().resolve("toolchain/Arduino-toolchain.cmake"),
-                    true
-                );
-                StringBuilder s = new StringBuilder();
-                s.append("set(ARDUINO_BOARD \"");
-                s.append("\")");
-                FileUtil.writeToFile(s.toString(), fileConfig.getSrcGenPath().resolve("toolchain/BoardOptions.cmake"));
-            }
-
-
-                // If Zephyr target then copy default config and board files 
-                //  for Zephyr support
-                if (targetConfig.platformOptions.platform == Platform.ZEPHYR) {
```

Comment:
The removal of the following lines causes the CI failure for Zephyr.

Reply:
Brought these lines back. Let's see what happens in CI :-)

Reply:
Looks like that worked. Thanks, @erlingrj!